rpc/src/main/scala/client/handlers/ClientCallsMHandler.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle.rpc
+package client.handlers
+
+import freestyle.Capture
+import freestyle.async.AsyncContext
+import freestyle.rpc.client.ClientCallsM
+import io.grpc.{CallOptions, Channel, ClientCall, MethodDescriptor}
+import io.grpc.stub.{ClientCalls, StreamObserver}
+import freestyle.rpc.client.implicits._
+
+import scala.collection.JavaConverters._
+
+class ClientCallsMHandler[F[_]](implicit C: Capture[F], AC: AsyncContext[F])
+    extends ClientCallsM.Handler[F] {
+
+  def asyncUnaryCall[I, O](
```

Comment:
Sounds good, as we have to differentiate internally (and `@free` doesn't support method overloading), these are the new names: `async`, `asyncStreamServer`, `asyncStreamClient`, `asyncStreamBidi`, `sync`, `syncC`, `syncStreamServer`, `syncStreamServerC`, `asyncM`